javascript/i18n/phonenumbers/demo.js

```diff
@@ -63,57 +64,68 @@ function phoneNumberParser() {
         case PNV.TOO_LONG:
           output.append('TOO_LONG');
           break;
+        case PNV.INVALID_LENGTH:
+          output.append('INVALID_LENGTH');
+          break;
       }
       // IS_POSSIBLE shouldn't happen, since we only call this if _not_
       // possible.
       output.append('\nNote: numbers that are not possible have type ' +
           'UNKNOWN, an unknown region, and are considered invalid.');
     } else {
-      var isNumberValid = phoneUtil.isValidNumber(number);
-      output.append('\nResult from isValidNumber(): ');
-      output.append(isNumberValid);
-      if (isNumberValid && regionCode && regionCode != 'ZZ') {
+      // Checking as isValid() fails if possible local only.
+      if (isPossibleReason == PNV.IS_POSSIBLE_LOCAL_ONLY) {
```

Comment:
Like the idea: but won't this not trigger, because this is when isPossible is true??

Reply:
As isPossible returns true for possible national and local numbers this is getting triggered.
https://screenshot.googleplex.com/BRXu6Lj3t2uuHwY

Reply:
but it is not in the "if" block where isPossible is true? It looks to me that it is in the else block?

Reply:
(your screenshot is out-of-date, because it doesn't have the message your code does)

Reply:
It is ofcourse in the block where isPossible is true. This is else block of
if(!isPossible) {...}
else {}

Reply:
Oh dear :) I kept reading if (isPossible) :)
Can we actually revert the if and else block order to make it clear??.......... sorry!!

Reply:
Np :) Made the update so it is clear.